crates/vizia_core/src/state/lens.rs

```diff
@@ -195,8 +220,11 @@ where
     type Source = A::Source;
     type Target = B::Target;
 
-    fn view<O, F: FnOnce(Option<&Self::Target>) -> O>(&self, source: &Self::Source, map: F) -> O {
-        self.a.view(source, |t| if let Some(t) = t { self.b.view(t, map) } else { map(None) })
+    fn view<'a>(&self, source: &'a Self::Source) -> Option<LensValue<'_, 'a, Self::Target>> {
+        match self.a.view(source)? {
+            LensValue::Borrowed(val) => self.b.view(val),
+            _ => None,
+        }
```

Comment:
There are more cases that are possible to handle here.
```suggestion
        let val = self.a.view(source)?;
        let val = match val {
            LensValue::Borrowed(val) => return self.b.view(val),
            LensValue::Local(val) => val,
            LensValue::Owned(ref val) => &val,
        };
        match self.b.view(&val)? {
            LensValue::Local(val) => Some(LensValue::Local(val)),
            LensValue::Owned(val) => Some(LensValue::Owned(val)),
            _ => None,
        }
```

Reply:
Why can’t we just store everything in Val and then do anything also for Borrow? Why these 2 extra rules for Borrow? (The direct calling and the exclude from the second match)

Reply:
I can't check right now but I assume it's because the lifetimes are different so it would be returning different types from the match, which isn't possible. I can double check that later though.